src/main/java/org/openjdk/jextract/impl/ClassSourceBuilder.java

```diff
@@ -50,160 +47,104 @@ enum Kind {
         }
     }
 
-    final Kind kind;
-    final ClassDesc desc;
-    protected final JavaSourceBuilder enclosing;
+    private final SourceFileBuilder sb;
+    private final List<String> enclosingClassNames;
+    private final String modifiers;
+    private final Kind kind;
+    private final String className;
+    private final String superName;
 
-    // code buffer
-    private StringBuilder sb = new StringBuilder();
-    // current line alignment (number of 4-spaces)
-    private int align;
-
-    ClassSourceBuilder(JavaSourceBuilder enclosing, Kind kind, String name) {
-        this.enclosing = enclosing;
-        this.align = (enclosing instanceof ClassSourceBuilder classSourceBuilder)
-                ? classSourceBuilder.align : 0;
+    ClassSourceBuilder(SourceFileBuilder builder, String modifiers, Kind kind, String className, String superName, List<String> enclosingClassNames) {
```

Comment:
If this models a class, and classes can be nested, it is not clear to me as to why we wouldn't have a pointer to the enclosing class, rather than having just a list of names?

Reply:
Yeah, that might be simpler. I started out trying to remove the reference to enclosing classes altogether, but that turned out to be impossible since we might need to reference a nested class from source code we generate.